### PR TITLE
security: address slither findings and add slither CI

### PR DESCRIPTION
**Pull request by adu-web3**
## Description 

closes: #37

Before the audit of contract codebase starts, we should audit by ourselves to make sure it is ready for audit. One of the approaches is by applying some static analyzing tools like slither to help us identify and fix the security issues. While some of the findings of slither are true positives, there are some false positives that should be ignored. 

we have two approaches to ignore false positives:
1. add exclude-detectors to `slither.config.json`
2. add `slither-disable-next-line` comment to the code that cause the issue

while approach 1 is easy, we should try approach 2 best because it would not disable the detector totally.

- [x] address valid findings: remove unused state variables, use cached array length...
- [x] add `slither-disable-next-line` to specific false positives
- [x] add exclude-detectors to `slither.config.json` for not needed detectors
- [x] add slither CI

## Summary by CodeRabbit

- **New Features**
  - Added support for new constant state variables for `TOKEN_ADDRESS_BYTES_LENGTH`, `DESTINATION_GAS_LIMIT`, and `DESTINATION_MSG_VALUE`.

- **Bug Fixes**
  - Corrected the import paths for `IBeacon` to ensure proper reference to external libraries.

- **Refactor**
  - Updated the `VALIDATOR_TREE_HEIGHT` to 40 and adjusted related constants.
  - Updated Solidity version pragma for consistency across files.

- **Chores**
  - Added a comment directive for `encode-packed-collision` in `_deployVault` function for improved code quality.

### CODE REVIEW

**Review by bwhour — APPROVED** [review]

**Review by MaxMustermann2 — APPROVED** [review]
Good job

File: src/core/Vault.sol [hunk 1]
```diff
@@ -49,6 +49,9 @@
         emit WithdrawalSuccess(withdrawer, recipient, amount);
     }
 
+    // Though `safeTransferFrom` has arbitrary passed in `depositor` as sender, this function is only callable by `gateway`
+    // and `gateway` would make sure only the `msg.sender` would be the depositor.
+    // slither-disable-next-line arbitrary-send-erc20
```
[the remark this answers is not in the document]
Comment (adu-web3) [reply src/core/Vault.sol]: fixed in new commits

File: src/storage/BootstrapStorage.sol [hunk 2]
```diff
@@ -452,6 +452,9 @@
         return true;
     }
 
+    // The bytecode returned by `BEACON_PROXY_BYTECODE` and `EXO_CAPSULE_BEACON` address are actually fixed size of byte array,
+    // so it would not cause collision for encodePacked
+    // slither-disable-next-line encode-packed-collision
```
[the remark this answers is not in the document]
Comment (adu-web3) [reply src/storage/BootstrapStorage.sol]: fixed in new commits

File: src/core/NativeRestakingController.sol [hunk 3]
```diff
@@ -45,6 +45,9 @@
         emit StakedWithCapsule(msg.sender, address(capsule));
     }
 
+    // The bytecode returned by `BEACON_PROXY_BYTECODE` and `EXO_CAPSULE_BEACON` address are actually fixed size of byte array,
+    // so it would not cause collision for encodePacked
+    // slither-disable-next-line encode-packed-collision
```
[the remark this answers is not in the document]
Comment (adu-web3) [reply src/core/NativeRestakingController.sol]: fixed in new commits

**Pull request merged**